### PR TITLE
par2cmdline-turbo: init at 1.1.1

### PR DESCRIPTION
**Pull request by proglottis**
## Description of changes

This is a fork of par2cmdline that adds performance improvements. This package can be used with sabnzbd to improve verification and repair performance.

* Homepage https://github.com/animetosho/par2cmdline-turbo
* Sab docs: https://sabnzbd.org/wiki/installation/par2cmdline-turbo

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/par2cmdline-turbo/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    sha256 = "sha256-+p6EeWn+tgZFRWSK4feQ4TIXJP+2iIcu9Ng0T6JNRCE=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+
+  meta = with lib; {
+    homepage = "https://github.com/animetosho/par2cmdline-turbo";
+    description = "par2cmdline × ParPar: speed focused par2cmdline fork";
+    license = licenses.gpl2Plus;
+    maintainers = [ ];
```
Comment (Lurkki14) [inline pkgs/tools/networking/par2cmdline-turbo/default.nix:20]: You need to add yourself as a maintainer https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

**Comment by Lurkki14** [comment]
The commit message for adding the package should be `par2cmdline-turbo: init at 1.1.0`   https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

File: pkgs/tools/networking/par2cmdline-turbo/default.nix (hunk 1 above)
Comment (proglottis) [reply pkgs/tools/networking/par2cmdline-turbo/default.nix]: Added

**Comment by Lurkki14** [comment]
The commit message for adding yourself to maintainers should be `maintainers: add proglottis`

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -11859,6 +11859,8 @@ with pkgs;
 
   par2cmdline = callPackage ../tools/networking/par2cmdline { };
 
+  par2cmdline-turbo = callPackage ../tools/networking/par2cmdline-turbo { };
+
```
Comment (Lurkki14) [inline pkgs/top-level/all-packages.nix:11863]: New packages are recommended to go to `pkgs/by-name` https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md

File: pkgs/tools/networking/par2cmdline-turbo/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "par2cmdline-turbo";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "animetosho";
+    repo = "par2cmdline-turbo";
+    rev = "v${version}";
+    sha256 = "sha256-+p6EeWn+tgZFRWSK4feQ4TIXJP+2iIcu9Ng0T6JNRCE=";
```
Comment (Lurkki14) [inline pkgs/tools/networking/par2cmdline-turbo/default.nix:11]:
```suggestion
    hash = "sha256-+p6EeWn+tgZFRWSK4feQ4TIXJP+2iIcu9Ng0T6JNRCE=";
```
The hash string already describes the algorithm

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (proglottis) [reply pkgs/top-level/all-packages.nix]: Moved. Thanks

File: pkgs/tools/networking/par2cmdline-turbo/default.nix (hunk 3 above)
Comment (proglottis) [reply pkgs/tools/networking/par2cmdline-turbo/default.nix]: Fixed

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -11861,6 +11861,8 @@ with pkgs;
 
   par2cmdline = callPackage ../tools/networking/par2cmdline { };
 
+  par2cmdline-turbo = callPackage ../by-name/pa/par2cmdline-turbo/package.nix { };
+
```
Comment (Lurkki14) [inline pkgs/top-level/all-packages.nix:11865]:
```suggestion
```
This isn't needed with the package being in `pkgs/by-name`

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (proglottis) [reply pkgs/top-level/all-packages.nix]: :heavy_check_mark: thanks

**Review by Lurkki14 — APPROVED** [review]

**Review by Nanotwerp — APPROVED** [review]
LGTM! Let's get this upgraded to v1.1.1, though.

**Comment by onemoresuza** [comment]
# Reviewed points

Result of `nixpkgs-review pr 264706` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>par2cmdline-turbo</li>
  </ul>
</details>

- [X] package path fits guidelines
- [X] package name fits guidelines
- [X] package version fits guidelines
- [X] package builds on x86_64-linux
- [X] executables tested on x86_64-linux
- [X] `meta.description` is set and fits guidelines
- [X] `meta.license` fits upstream license
- [X] `meta.platforms` is set
- [X] `meta.maintainers` is set
- [X] build time only dependencies are declared in `nativeBuildInputs`
- [X] source is fetched using the appropriate function
- [X] the list of `phases` is not overridden
- [ ] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
	- N/A: No phase is overridden
- [ ] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
	- N/A: No patch is applied 
- [ ] patches that are remotely available are fetched rather than vendored
	- N/A: No patch is applied 

# Possible improvements

## Commit message does not adhere to conventions

As stated by @Lurkki14, the commit message should be `par2cmdline-turbo: init at
1.1.0`, that is, without the `v` prefix, as it is shown in the [commit messages
conventions for pkgs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions).

## Parallel building

I've added `enableParallelBuilding = true;` to the derivation and the project
built without any issue. Any reason not to enable it?

## New upstream version

As stated by @Nanotwerp, upstream has released a [new version](https://github.com/animetosho/par2cmdline-turbo/releases/tag/v1.1.1), so this PR could bump its version to newest one.

## `mainProgram` meta attribute

The derivation contains only one binary (`par2`), the other files in `bin/` are just symlinks to it. Hence, it could become the projects [`meta.mainProgram`](https://nixos.org/manual/nixpkgs/stable/#var-meta-mainProgram).

File: pkgs/by-name/pa/par2cmdline-turbo/package.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "animetosho";
+    repo = "par2cmdline-turbo";
+    rev = "v${version}";
+    hash = "sha256-+p6EeWn+tgZFRWSK4feQ4TIXJP+2iIcu9Ng0T6JNRCE=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+
```
Comment (onemoresuza) [inline pkgs/by-name/pa/par2cmdline-turbo/package.nix:15]:
```suggestion

enableParallelBuilding = true;
```

File: pkgs/by-name/pa/par2cmdline-turbo/package.nix [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  meta = with lib; {
+    homepage = "https://github.com/animetosho/par2cmdline-turbo";
+    description = "par2cmdline × ParPar: speed focused par2cmdline fork";
+    license = licenses.gpl2Plus;
+    maintainers = [ maintainers.proglottis ];
+    platforms = platforms.all;
```
Comment (onemoresuza) [inline pkgs/by-name/pa/par2cmdline-turbo/package.nix:21]:
```suggestion
    platforms = platforms.all;
    mainProgram = "par2";
```

**Review by onemoresuza — CHANGES_REQUESTED** [review]

File: pkgs/by-name/pa/par2cmdline-turbo/package.nix (hunk 5 above)
Comment (proglottis) [reply pkgs/by-name/pa/par2cmdline-turbo/package.nix:16]: Applied, thanks

File: pkgs/by-name/pa/par2cmdline-turbo/package.nix (hunk 6 above)
Comment (proglottis) [reply pkgs/by-name/pa/par2cmdline-turbo/package.nix:22]: Added

**Comment by onemoresuza** [comment]
Result of `nixpkgs-review pr 264706` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>par2cmdline-turbo</li>
  </ul>
</details>

**Review by onemoresuza — APPROVED** [review]

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Pull request merged**